web/src/filter.js

```diff
@@ -501,6 +501,26 @@ export class Filter {
     }
 
     calc_can_mark_messages_read() {
+        if (this.all_messages_of_conversation_included()) {
+            return true;
+        }
+        const term_types = this.sorted_term_types();
+        if (_.isEqual(term_types, [])) {
+            // All view
+            return true;
```

Comment:
// All view not needed inside is_common_narrow because all messages narrow has a filter as undefined and therefore we are already handling that case everywhere.

Reply:
Okay. I see that you've worked on explaining this in the first commit message. But that commit message is very difficult to read and understand as it is. I suggest rewording it for clarity and simplicity.

Here's a go at a revision from me:

```console
filter: Create helper function that checks basic filter terms.

Currently, `is_common_narrow` relies on `calc_can_mark_messages_read`
to check a number of filters that are "common narrows" (which means
they have a special message header title and different behavior when
exiting a search view).

Creates a helper function, `all_messages_of_conversation_included`,
that is used in both the above functions.

The check for an empty array of term types is unnecessary for
`is_common_narrow` because the "All messages" view has an undefined
filter and the "in-home" term type used in the "zhome" message list
is covered. So that empty array check is not moved to the helper
function, and instead stays only in `calc_can_mark_messages_read`.

The helper function checks for a single term type (except in the case
of topic, which is checked in combination with stream) that will return
all the messages of a conversation.
```

Reply:
Done